Core/include/Acts/Propagator/EigenStepper.ipp

```diff
@@ -233,8 +233,41 @@ Acts::Result<double> Acts::EigenStepper<E, A>::step(
       return EigenStepperError::StepInvalid;
     }
 
-    // for moment, only update the transport part
-    state.stepping.jacTransport = D * state.stepping.jacTransport;
+    // See the documentation of Acts::blockedMult for a description of blocked
+    // matrix multiplication. However, we can go one further. Let's assume that
+    // some of these sub-matrices are zero matrices 0₈ and identity matrices
+    // I₈, namely:
+    //
+    // D₁₁ = I₈, J₁₁ = I₈, D₂₁ = 0₈, J₂₁ = 0₈
+    //
+    // Which gives:
+    //
+    // K₁₁ = I₈  * I₈  + D₁₂ * 0₈  = I₈
+    // K₁₂ = I₈  * J₁₂ + D₁₂ * J₂₂ = J₁₂ + D₁₂ * J₂₂
+    // K₂₁ = 0₈  * I₈  + D₂₂ * 0₈  = 0₈
+    // K₂₂ = 0₈  * J₁₂ + D₂₂ * J₂₂ = D₂₂ * J₂₂
+    //
+    // Furthermore, we're constructing K in place of J, and since
+    // K₁₁ = I₈ = J₁₁ and K₂₁ = 0₈ = D₂₁, we don't actually need to touch those
+    // sub-matrices at all!
+    if ((D.topLeftCorner<4, 4>().isIdentity(1e-8)) &&
+        (D.bottomLeftCorner<4, 4>().isZero(1e-8)) &&
+        (state.stepping.jacTransport.template topLeftCorner<4, 4>().isIdentity(
+            1e-8)) &&
+        (state.stepping.jacTransport.template bottomLeftCorner<4, 4>().isZero(
+            1e-8))) {
```

Comment:
I am a bit worried about the hardcoded thresholds

Reply:
We can give it another try without them.